src/ipc/ipc4/helper.c

```diff
@@ -779,8 +787,14 @@ __cold int ipc_comp_disconnect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	pipeline_disconnect(src, buffer, PPL_CONN_DIR_COMP_TO_BUFFER);
 	pipeline_disconnect(sink, buffer, PPL_CONN_DIR_BUFFER_TO_COMP);
 	/* these might call comp_ipc4_bind_remote() if necessary */
-	ret = comp_unbind(src, bu);
-	ret1 = comp_unbind(sink, bu);
+	unbind_data.ipc4_data = bu;
+	unbind_data.from_source = NULL;
+	unbind_data.from_sink = audio_buffer_get_sink(&buffer->audio_buffer);
+	ret = comp_unbind(src, &unbind_data);
+	
+	unbind_data.from_sink = NULL;
+	unbind_data.from_source = audio_buffer_get_source(&buffer->audio_buffer);
+	ret1 = comp_unbind(sink, &unbind_data);
```

Comment:
As an earlier comment from @kv2019i - we do have a lot of structure packing that could be done via a macro or static inline for readability.